src/query/interpreter.cpp

```diff
@@ -3790,6 +3791,9 @@ void RunTriggersIndividually(const utils::SkipList<Trigger> &triggers, Interpret
     auto storage_acc = interpreter_context->db->Access();
     DbAccessor db_accessor{storage_acc.get()};
 
+    // On-disk storage removes all Vertex/Edge Accessors after DbAccessor is destroyed.
+    // So we need to adapt TriggerContext based on user transaction which is still alive.
+    auto trigger_context = original_trigger_context;
```

Comment:
Can you explain me why this copying works?

Reply:
We talked and it makes sense! Do you think it would be good to add check if we are on disk storage since for in-memory it will make unnecessary copies?